pkg/internal/sqlsmith/scalar.go

```diff
@@ -373,6 +379,76 @@ func makeFunc(s *scope, ctx Context, typ types.T, refs colRefs) (tree.TypedExpr,
 	), typ), true
 }
 
+const maxWindowFrameOffset = 10
+
+func makeWindowFrame() *tree.WindowFrame {
+	// Window frame mode and start bound must always be present whereas end
+	// bound can be omitted.
+	frameMode := tree.WindowFrameMode(rand.Intn(3))
```

Comment:
Use the pattern similar to [randDirection](https://github.com/cockroachdb/cockroach/blob/01c915265aacdd07738591831af1b24ec82fa2fe/pkg/internal/sqlsmith/relational.go#L281) for this. You'll have to pass this func a `s *scope` argument. I guess BoundType is fine because of how you need to do the slicing for endBounds.